packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -438,13 +438,39 @@ export function createInsightsMiddleware<
           };
         }
 
+        const viewedObjectIDs = new Set<string>();
+        let lastQueryId: string | undefined;
+        instantSearchInstance.mainHelper!.derivedHelpers[0].on(
+          'result',
+          ({ results }) => {
+            if (!results.queryID || results.queryID !== lastQueryId) {
+              lastQueryId = results.queryID;
```

Comment:
I guess this works with multi-index too as if one index changes, they all change?

Reply:
Yes :)